modules/solid_mechanics/src/materials/SolidModel.C

```diff
@@ -1596,6 +1596,11 @@ SolidModel::createConstitutiveModel(const std::string & cm_name)
 
   params.applyParameters(parameters());
   params.set<SubProblem *>("_subproblem") = &_subproblem;
+  if (parameters().isParamValid("_material_data_type"))
+    params.set<Moose::MaterialDataType>("_material_data_type") =
+        getParam<Moose::MaterialDataType>("_material_data_type");
+  if (parameters().isParamValid("_neighbor"))
+    params.set<bool>("_neighbor") = getParam<bool>("_neighbor");
```

Comment:
I don't think I like this fix...  From looking at this tiny piece of code, I'm assuming you are being bitten by rule 4 in the applyParameters() routine (i.e. both (parameters) cannot be private). I can appreciate the logic that @aeslaughter used in building that routine but if it's going to cause issues in practice, perhaps we need to rethink that logic a bit? 

Private parameters often hold things like pointers to members so we probably shouldn't just go and copy them. Maybe one side or the other shouldn't have one of these parameters listed as private? 

Maybe we should have an override for applyParameters that will apply private parameters? I worry that this kind of workaround will bleed further out than we want it to. Any ideas?

Reply:
How about adding a param specific version of applyParameters:

```C++
params.applyParameters(parameters(), {"_material_data_type", "_neighbor"});
```

Reply:
I can add this feature if @permcody likes the idea

Reply:
Cody, I don't think we should perform any framework fixes to make this more elegant. This pattern (factory calls in a material) is limited to solid mechanics and will soon die along with it. The change that Ben implemented in the previous PR helped clean up some very misleading input files and was intended as a short term fix to make sure current users are getting out of their simulations what they intended. 

I am not in love with this fix, but it is a consequence of this undesirable pattern of creating unregistered material objects. I would strongly advise against accommodating this pattern and would suggest to permit this fix given that it has an expiration date.

Reply:
@dschwen - OK. Feel free to merge.
@aeslaughter - let's not worry about it then until we have a better use case where there's a compelling reason.

Reply:
Yeah, I'll be the first to admit that this is ugly.  I'm also hoping that we can delete this code really soon!